media_driver/linux/common/os/i915/mos_bufmgr.c

```diff
@@ -2012,7 +2238,7 @@ mos_gem_get_pipe_from_crtc_id(struct mos_bufmgr *bufmgr, int crtc_id)
 static void
 mos_gem_bo_wait_rendering(struct mos_linux_bo *bo)
 {
-    mos_gem_bo_start_gtt_access(bo, 1);
+    mos_gem_bo_wait(bo, -1);
```

Comment:
Endless recursion if the kernels !bufmgr_gem->has_wait_timeout?

Reply:
:) Yeah, you are right. I missed that.

But this also means we can't fully switch to gem_wait here since on older kernel we might not have it and there is no alternative other than to call set_domain. For what I understand mos_gem_bo_busy is non-blocking function - is that right?

So, right now I intend to simply do:
```
if (!bufmgr_gem->has_wait_timeout) {
    mos_gem_bo_start_gtt_access(bo, 1);
} else {
    memclear(wait);
    wait.bo_handle = bo_gem->gem_handle;
    wait.timeout_ns = -1;
    drmIoctl(bufmgr_gem->fd, DRM_IOCTL_I915_GEM_WAIT, &wait);
}
```

How do you think?

Reply:
done per proposal above